solr/modules/opentelemetry/build.gradle

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply plugin: 'java-library'
+
+description = 'Open Telemetry (OTEL) tracer'
+
+// This is a hacky way to use permitTestUnusedDeclared with bom declared dependencies.
+// See https://github.com/gradle-dependency-analyze/gradle-dependency-analyze/issues/108
+configurations {
+  constraintsOnly
+  permitUnusedDeclared.extendsFrom constraintsOnly
+  implementation.extendsFrom constraintsOnly
+  runtimeOnly.extendsFrom constraintsOnly
+}
+
+dependencies {
+  implementation project(':solr:core')
+
+  constraintsOnly(platform('io.opentelemetry:opentelemetry-bom-alpha'))
+
+  implementation 'io.opentracing:opentracing-api'
+  implementation 'org.slf4j:slf4j-api'
+
+  implementation 'io.opentelemetry:opentelemetry-sdk'
+  implementation 'io.opentelemetry:opentelemetry-sdk-trace'
+  implementation 'io.opentelemetry:opentelemetry-sdk-extension-autoconfigure'
+  implementation 'io.opentelemetry:opentelemetry-opentracing-shim'
+  runtimeOnly 'io.opentelemetry:opentelemetry-semconv'
+  permitUnusedDeclared 'io.opentelemetry:opentelemetry-semconv'
+  runtimeOnly 'io.opentelemetry:opentelemetry-exporter-otlp'
+  permitUnusedDeclared 'io.opentelemetry:opentelemetry-exporter-otlp'
+  // End users must recompile with jaeger exporter and/or zipkin exporter if they need these
+
+  // NOTE: sdk-autoconfigure needs both opentelemetry-sdk-metrics and opentelemetry-sdk-logs even if we don't use them
+
+  // gRPC transport via netty - since we already ship netty this is more lightweight than netty-shaded
+  runtimeOnly 'io.grpc:grpc-netty'
+  permitUnusedDeclared 'io.grpc:grpc-netty'
```

Comment:
If these are runtimeOnly, I'm 90% sure that we don't need `permitUnusedDeclared` anymore. 

I think this means that `permitUnusedDeclared` isn't needed can be removed from the entire build.gradle and the section for

```
configurations {
  constraintsOnly
  permitUnusedDeclared.extendsFrom constraintsOnly
  implementation.extendsFrom constraintsOnly
  runtimeOnly.extendsFrom constraintsOnly
}
```

which is just to handle these bom unused declared can be removed as well?

Reply:
Yey, after moving those to runtimeOnly above, we no longer need `permitUnusedDeclared`. I removed it all, and moved the bom declaration to `runtimeOnly`, would that be correct? gradle check does not complain...

Reply:
added a separate comment but - `platform('io.opentelemetry:opentelemetry-bom-alpha')` should do it.